betterproto/__init__.py

```diff
@@ -443,25 +522,20 @@ def __post_init__(self) -> None:
 
         # Set a default value for each field in the class after `__init__` has
         # already been run.
-        group_map: Dict[str, dict] = {"fields": {}, "groups": {}}
+        group_map: Dict[str, dataclasses.Field] = {}
         for field in dataclasses.fields(self):
             meta = FieldMetadata.get(field)
 
             if meta.group:
-                # This is part of a one-of group.
-                group_map["fields"][field.name] = meta.group
-
-                if meta.group not in group_map["groups"]:
-                    group_map["groups"][meta.group] = {"current": None, "fields": set()}
-                group_map["groups"][meta.group]["fields"].add(field)
+                group_map.setdefault(meta.group)
```

Comment:
Can you help me grok this? is this the same result before and after, or does this dict have very different structure with this change?

From what I can tell, some keys are now removed. There is no concern for these groups being overwritten? or the possibility is not there because such behavior cannot occur in the protocol buffer that this will process. Is that right?

Reply:
Originally, the `group_map` is like:

```python
{
    "fields": {
        "field1": "group1",
        "field2": "group1",
    },
    "groups": {
        "group1": {
            "current": Field1,
            "fields": set(Field1, Field2),
        },
    },
}
```
Within this dict, only `current` is object scoped, others are all class metadata. So that I kept `current` in this dict and moved others into `ProtoClassMetadata`, as `oneof_group_by_field` and `oneof_field_by_group`. Please refer to method `__setattr__`, L555 of changed `__init__.py` file.

In this way, building the lookup table between fields and groups is executed once per class. And also, all the object s of the class share the same lookup table.

Now in the object, `_group_map` indicates the current value of oneof group:

```python
{
    "group1": Field1,
}
```

`__setattr__` will keep this updated